dts/bindings/test/vnd,adc-temp-sensor.yaml

```diff
@@ -21,3 +21,9 @@ properties:
 
     clocks:
       required: true
+
+    pinctrl-0:
+      type: phandles
+
+    pinctrl-1:
+      type: phandles
```

Comment:
should we provide something like this: https://github.com/zephyrproject-rtos/zephyr/blob/ca4c7d200ee6f1ef77dbfe1ffef8be537af140da/dts/bindings/pinctrl/pinctrl-device.yaml ?

Reply:
@mbolivar-nordic I don't know if you want to add `pinctrl-0` & `pinctrl-1`, and `pinctrl-names` to dts/bindings/base/base.yaml as part of this PR.

Reply:
I think I'll just include pinctrl-device.yaml into vnd,adc-temp-sensor for now. I think base.yaml is pretty big already and it's hard for users to know when a property in there is relevant to their device or not.

Reply:
@gmarull actually, that commit doesn't seem to be upstream yet...

Reply:
It's part of #37572, feel free to take that file.

Reply:
> It's part of #37572, feel free to take that file.

Thanks but no thanks. Pulling that file in without updating all the in-tree bindings to include it feels wrong to me, and that's outside the scope of this PR.